docs/hub/datasets-pandas.md

```diff
@@ -22,26 +57,163 @@ Finally, you can use [Hugging Face paths](/docs/huggingface_hub/guides/hf_file_s
+
+```
+Example 1:            Example 2:
+folder/               folder/
+├── metadata.csv      ├── metadata.csv
+├── img000.png        └── images
+├── img001.png            ├── img000.png
+...                       ...
+└── imgNNN.png            └── imgNNN.png
+```
+
+You can iterate on the images paths like this:
+
+```python
+import pandas as pd
+
+folder_path = "path/to/folder/"
+df = pd.read_csv(folder_path + "metadata.csv")
+for image_path in (folder_path + df["file_name"]):
+    ...
+```
+
+Since the dataset is in a supported structure ("metadata.csv" file with "file_name" field), you can save this dataset to Hugging Face and the Dataset Viewer shows both the metadata and images on Hugging Face.
+
+```python
+from huggingface_hub import HfApi
+api = HfApi()
+
+api.upload_folder(
+    folder_path=folder_path,
+    repo_id="username/my_image_dataset",
+    repo_type="dataset",
+)
+```
+
+### Image methods and Parquet
+
+Using [pandas-image-methods](https://github.com/lhoestq/pandas-image-methods) you enable `PIL.Image` methods on an image column. It also enables saving the dataset as one single Parquet file containing both the images and the metadata:
 
 ```python
 import pandas as pd
+from pandas_image_methods import PILMethods
 
-df = pd.read_parquet("hf://datasets/username/my_dataset/data.parquet")
+pd.api.extensions.register_series_accessor("pil")(PILMethods)
 
-# or read from separate files if the dataset has train/validation/test splits
-df_train = pd.read_parquet("hf://datasets/username/my_dataset/train.parquet")
-df_valid = pd.read_parquet("hf://datasets/username/my_dataset/validation.parquet")
-df_test  = pd.read_parquet("hf://datasets/username/my_dataset/test.parquet")
+df["image"] = (folder_path + df["file_name"]).pil.open()
+df.to_parquet("data.parquet")
 ```
 
-To have more information on the Hugging Face paths and how they are implemented, please refer to the [the client library's documentation on the HfFileSystem](/docs/huggingface_hub/guides/hf_file_system).
+All the `PIL.Image` methods are available, e.g.
+
+```python
+df["image"] = df["image"].pil.rotate(90)
+```
+
+## Use Audios
+
+You can load a folder with a metadata file containing a field for the names or paths to the audios, structured like this:
+
+```
+Example 1:            Example 2:
+folder/               folder/
+├── metadata.csv      ├── metadata.csv
+├── rec000.wav        └── audios
+├── rec001.wav            ├── rec000.wav
+...                       ...
+└── recNNN.wav            └── recNNN.wav
+```
+
+You can iterate on the audios paths like this:
+
+```python
+import pandas as pd
+
+folder_path = "path/to/folder/"
+df = pd.read_csv(folder_path + "metadata.csv")
+for audio_path in (folder_path + df["file_name"]):
+    ...
+```
+
+Since the dataset is in a supported structure ("metadata.csv" file with "file_name" field), you can save this dataset to Hugging Face and the Dataset Viewer shows both the metadata and audios on Hugging Face.
```

Comment:
```suggestion
Since the dataset is in a supported structure (a `metadata.csv` file with a `file_name` field), you can save it to Hugging Face, and the Hub Dataset Viewer shows both the metadata and audio. 
```